tasks/managed/push-rpm-to-koji/push-rpm-to-koji.yaml

```diff
@@ -0,0 +1,150 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: push-rpm-to-koji
+spec:
+  params:
+    - name: snapshotPath
+      type: string
+      description: |
+        Path to the JSON string of the mapped Snapshot spec in the data workspace.
+    - name: dataPath
+      type: string
+      description: |
+        Path to the JSON string of the merged data to use in the data workspace.
+    - name: pushSecret
+      type: string
+      description: |
+        The secret that is used for login koji instance.
+    - name: subdirectory
+      type: string
+      description: |
+        The pipeline specific directory.
+    - name: pipelineImage
+      type: string
+      description: |
+        The image url with koji, kinit installed for running the push-rpm-to-koji task,
+        please make sure you have such image or you build this image first.
+  steps:
+    - name: push-rpm-to-koji
+      # The pipelineImage will be fetching from get-secrets task with koji installed there
+      image: "$(params.pipelineImage)"
+      script: |
+        #!/usr/bin/env bash
+        set -eux
+
+        SNAPSHOT_SPEC_FILE="$(workspaces.data.path)/$(params.snapshotPath)/snapshot_spec.json"
+        if [ ! -f "${SNAPSHOT_SPEC_FILE}" ] ; then
+            echo "No valid snapshot file was provided."
+            exit 1
+        fi
+
+        DATA_FILE="$(workspaces.data.path)/$(params.dataPath)/data.json"
+        if [ ! -f "${DATA_FILE}" ] ; then
+            echo "No data JSON was provided."
+            exit 1
+        fi
+
+        APPLICATION=$(jq -r '.application' "${SNAPSHOT_SPEC_FILE}")
+        NUM_COMPONENTS=$(jq '.components | length' "${SNAPSHOT_SPEC_FILE}")
+
+        PRINCIPLE=$(jq -r '.pushOptions.pushKeytab.principle' "${DATA_FILE}")
```

Comment:
it is principal for keytabs, not principle, no?

Reply:
The structure is like: 
"pushKeytab": {
  "name": "test.keytab",
  "principle": "test@test.com",
  "secret": "test-secrets"
},

Reply:
Oh, thank you, changing...